src/Xrefcheck/Scanners/Markdown.hs

```diff
@@ -217,28 +210,47 @@ removeIgnored fp = withIgnoreMode . cataNodeWithParentNodeInfo remove
 
       return scan
 
-    handleIgnoreMode
+    handleAnnotation
       :: Maybe PosInfo
       -> NodeType
-      -> [ScannerM Node]
-      -> GetIgnoreMode
+      -> GetAnnotation
       -> ScannerM Node
-    handleIgnoreMode pos nodeType subs = \case
-      ValidMode mode  -> do
-        ignoreModeState <- case mode of
-          IMLink -> use ssParentNodeType <&> IMSLink . \case
-             Just PARAGRAPH -> ExpectingLinkInParagraph
-             _ -> ExpectingLinkInSubnodes
-
-          IMParagraph -> pure IMSParagraph
-
-          IMAll -> pure IMSAll
-
-        (ssIgnore .= Just (Ignore ignoreModeState correctPos)) $> defNode
-      InvalidMode msg -> do
+    handleAnnotation pos nodeType = \case
+      IgnoreAnnotation mode  -> do
+        let reportIfThereWasAnnotation :: ScannerM ()
+            reportIfThereWasAnnotation = do
+              curIgnore <- use ssIgnore
+              whenJust curIgnore $ \case
+                Ignore IMSParagraph prevPos ->
+                  lift . tell . makeError prevPos fp . ParagraphErr $ prettyType nodeType
+                Ignore (IMSLink _) prevPos ->
+                  lift $ tell $ makeError prevPos fp LinkErr
```

Comment:
Here too, the commit says that you apply a refactoring, but here in fact you add a behaviour that was not present before.

I more than agree that this check is good to add, but let's extract it to a separate commit.

Reply:
In fact the behavior hasn't changed here, because in the original version some of these checks were present in other cases on the top level of `remove` function.